src/extra/others/controller/lv_obj_controller.c

```diff
@@ -0,0 +1,372 @@
+/**
+ * @file lv_obj_controller.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "lv_obj_controller.h"
+
+#if LV_USE_OBJ_CONTROLLER
+
+#include "../../../extra/widgets/msgbox/lv_msgbox.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef struct manager_stack_t {
+    const lv_obj_controller_class_t * cls;
+    lv_obj_controller_t * controller;
+    bool obj_created;
+    bool destroying_obj;
+    bool dialog;
+    struct manager_stack_t * prev;
+} manager_stack_t;
+
+struct lv_controller_manager_t {
+    lv_obj_t * container;
+    lv_obj_controller_t * parent;
+    manager_stack_t * top;
+};
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static manager_stack_t * item_new(const lv_obj_controller_class_t * cls);
+
+static lv_obj_controller_t * item_create_controller(lv_controller_manager_t * manager, manager_stack_t * item,
+                                                    void * args);
+
+static void item_create_obj(lv_controller_manager_t * manager, manager_stack_t * item, lv_obj_t * parent,
+                            const lv_obj_class_t * check_type);
+
+static void item_destroy_obj(lv_controller_manager_t * manager, manager_stack_t * item);
+
+static void item_destroy_controller(manager_stack_t * item);
+
+static void obj_cb_delete(lv_event_t * event);
+
+static void controller_destroy_obj(lv_obj_controller_t * controller);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+lv_controller_manager_t * lv_controller_manager_create(lv_obj_t * container, lv_obj_controller_t * parent)
```

Comment:
Which API function can return an `lv_obj_controller_t * ` that can be used as `parent`?

Reply:
Functions in controller lifecycle will have access to `lv_obj_controller_t *`. e.g. `lv_obj_controller_class_t.constructor_cb`.

Reply:
I mean how can I get pointer that can be passed as `parent`?

Reply:
like this:

```c
void create_obj_cb(struct lv_obj_controller_t * self, lv_obj_t * container) {
    my_controller * controller = self;
    controller->child_manager = lv_controller_manager_create(controller->container_obj, self);
}
```

Reply:
I see! So a manager can have only one child?

Reply:
Actually they can have whatever they want. Nested manager however, doesn't persist stack yet. If there is need then I'll implement in the future.

Reply:
> Actually they can have whatever they want. 

How can I add two children then?

Reply:
```c
void my_obj_created_cb(struct lv_obj_controller_t * self, lv_obj_t * container) {
    my_controller * controller = self;
    controller->child_manager1 = lv_controller_manager_create(controller->container_obj1, self);
    controller->child_manager2 = lv_controller_manager_create(controller->container_obj2, self);
}
```

Such use case can be found when there are multiple panes.